pyrokinetics/local_species.py

```diff
@@ -61,33 +60,20 @@ def from_dict(self, species_dict, **kwargs):
             super(LocalSpecies, self).__init__(*sort_species_dict, **kwargs)
 
     @classmethod
-    def from_global_kinetics(cls, kinetics: Kinetics, psi_n: float, lref: float):
+    def from_global_kinetics(
+        cls, kinetics: Kinetics, psi_n: float, local_norm: Normalisation
+    ):
         # TODO this should replace from_kinetics
         local_species = cls()
-        local_species.from_kinetics(kinetics, psi_n=psi_n, lref=lref)
+        local_species.from_kinetics(kinetics, psi_n=psi_n, local_norm=local_norm)
         return local_species
 
-    def from_kinetics(self, kinetics, psi_n=None, lref=None):
+    def from_kinetics(self, kinetics, psi_n, local_norm):
```

Comment:
I'm not actually convinced that passing in `local_norm` is correct here -- previously it was generated to be consistent with `kinetics`, although `lref` had to be passed in.

Maybe it should be optional, and if `None`, then it uses `kinetics` for the normalisation?

Reply:
`lref` isn't stored in `kinetics` so how else would we get it? We could directly pass `lref` by doing

```python
local_species.from_kinetics(kinetics, psi_n=psi_n, lref=local_norm.lref)
```